src/BloomBrowserUI/lib/localizationManager.ts

```diff
@@ -138,7 +140,69 @@ class LocalizationManager {
     return HtmlDecode(text);
   }
 
-  /**
+    /* Returns a promise to get the translation
+     * 
+     * @param {String} langId : can be an iso 639 code or one of these constants: UI, V, N1, N2
+     * 
+     *  @example
+     * asyncGetTextInLang('topics.health','Health", "UI")
+     *      .done(translation => {
+     *          $(this).text(translation);
+     *      })
+     *      .fail($(this).text("?Health?";
+     * @example
+     * asyncGetTextInLang('topics.health','My name is {0}", "UI", "John")
+     *      .done(translation => {
+     *          $(this).text(translation);
+     *      });
+
+    */
+    asyncGetTextInLang(id: string, englishText: string, langId: string, ...args): JQueryPromise {
+//        return getIframeChannel().asyncGet("/bloom/i18n/translate", { key: id, englishText: englishText, langId: langId });
+
+        // We already get a promise from the async call, and could just return that.
+        // But we want to first massage the data we get back from the ajax call, before we re - "send" the result along
+        //to the caller. So, we do that by making our *own* deferred object, and "resolve" it with the massaged value.
+        var deferred = new $.Deferred();
+        var promise = getIframeChannel().asyncGet("/bloom/i18n/translate", { key: id, englishText: englishText, langId: langId });
+        //when the async call comes back, we massage the text
+        promise.done(text => {
+            // is this a C#-style string.format style request?
+            if (args.length > 0) {
+                text = SimpleDotNetFormat(text, args);
+            }
+            deferred.resolve(HtmlDecode(text));
```

Comment:
Seems a comment got lost here, or maybe I never saved it...should we do the decode before the args test? I don't see any advantage to requiring the client to pass the extra args (which never become part of a URL or response) encoded.